qubes/tools/qubesd.py

```diff
@@ -15,6 +15,22 @@
 import qubes.utils
 import qubes.vm.qubesvm
 
+def init_random():
+    """
+    Wait for the system entropy pool to fill, so we can use ``/dev/urandom``
+    with confidence.
+    """
+    import ctypes
+    libc = ctypes.CDLL('libc.so.6', use_errno=True)
+    getentropy = libc['getentropy']
+    getentropy.restype = ctypes.c_int
+    getentropy.argtypes = (ctypes.c_void_p, ctypes.c_size_t)
+    buffer = ctypes.create_string_buffer(1)
+    if getentropy(buffer, 1):
+        err = ctypes.get_errno()
+        raise OSError(err, os.strerror(err))
```

Comment:
```suggestion
    with open('/dev/random', 'rb') as random:
        random.read(1)
```

Or `os.getrandom(1)` for Python 3.6+.

You could even skip this whole commit if the `cryptsetup` invocation is changed to use `/dev/random`, assuming [kernel 5.6+](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=30c08efec8884fb106b8e57094baa51bb4c44e32).

Reply:
I can’t assume kernel 5.6+, sadly.  But I *can* assume Python 3.6+ in R4.1.

Reply:
Yes, `os.getrandom(1)` would be better.